cfme/tests/containers/test_containers_overview_data_integrity.py

```diff
@@ -81,15 +75,42 @@ def test_containers_overview_data_integrity():
     navigate_to(ContainersOverview, 'All')
     # We should wait ~2 seconds for the StatusBox population
     # (until we find a better solution)
+    # Since we collect images from Openshift and from the pods,
+    # images are tested separately
     time.sleep(2)
     statusbox_values = {data_set.object: int(StatusBox(data_set.name).value())
                         for data_set in DATA_SETS}
-    api_values = get_api_object_counts(list_providers_by_class(ContainersProvider))
+    api_values = get_api_object_counts(
+        list_providers_by_class(ContainersProvider))
+
+    list_img_from_registry = provider.mgmt.list_image()
+    list_img_from_registry_splitted = [i.id.split(
+        '@sha256:')[-1] for i in list_img_from_registry]
+
+    list_img_from_openshift = provider.mgmt.list_image_openshift()
+    list_img_from_openshift_splitted = [d['name']
+                                        for d in list_img_from_openshift]
+    list_img_from_openshift_parsed = [i[7:].split(
+        '@sha256:')[-1] for i in list_img_from_openshift_splitted]
+
+    for s in list_img_from_openshift_parsed:
+        for item in list_img_from_registry_splitted:
+            if item not in list_img_from_openshift_parsed:
+                list_img_from_openshift_parsed.append(item)
+
+    assert len(list_img_from_openshift_parsed) == StatusBox('images').value()
+
+    list_all_rgstr = provider.mgmt.list_image_registry()
+    list_all_rgstr_revised = [i.host for i in list_all_rgstr]
+    list_all_rgstr_new = filter(lambda ch: 'openshift3' not in ch, list_all_rgstr_revised)
+
+    assert len(list_all_rgstr_new) == StatusBox('registries').value()
+
     results = {}
     for cls in DATA_SETS:
         results[cls.object] = api_values[cls.object] == statusbox_values[cls.object]
     if not all(results.values()):
         pytest.fail('There is a mismatch between API and UI values:\n{}'.format(
-                    '\n'.join(['{}: {} (API) != {} (UI)'.format(
-                        obj.__name__, api_values[obj], statusbox_values[obj])
-                        for obj, is_pass in results.items() if not is_pass])))
+            '\n'.join(['{}: {} (API) != {} (UI)'.format(
+                obj.__name__, api_values[obj], statusbox_values[obj])
+                for obj, is_pass in results.items() if not is_pass])))
```

Comment:
@gshefer Indeed, changed the nested loop to comprehension, will do the same in the other test